### ISSUE TITLE
EntropyBottleneck: make training flag compatible with TensorFlow Booleans.

### ISSUE THREAD

**Issue by msmsajjadi** [tensorflow/compression#14]
https://github.com/tensorflow/compression/blob/d156bbe3f85b332679e31bc0ace02e794ac044f8/tensorflow_compression/python/layers/entropy_models.py#L416-L423

The `EntropyBottleneck` layer currently only takes Python Booleans for its `training` flag. A more comfortable option would be to also accept TensorFlow Booleans so that the same layer can be used for evaluation of test-performance during training (similar to the `training` flag in Batch Normalization layers).

As far as I can see, adding in a `tf.cond` here is unproblematic and shouldn't hurt performance, but I may be missing something. I can submit a PR if that helps.

**Comment by jonycgn** [comment]
Yes, this is a limitation of the current design. We simply haven’t needed to use this feature internally, but I agree it’s a good thing to support.

We’re planning to release some new features soon, and I’ll look into supporting this with the next release.

**Comment by msmsajjadi** [comment]
Great, looking forward to the new features!

**Comment by jonycgn** [comment]
I'm going to reopen this to help me remember looking into it :)

**Comment by jonycgn** [comment]
Hi,

we're moving towards a new implementation of the entropy models for eager mode (see commit b996d5c277a3957b7be0c578e5d1bc4f12afe330 – these are going to be in the next release). These will make problems like this obsolete.

We're going to keep the old entropy models around, but they will only be supported in graph mode and phased out. Given that, I'm going to close this issue.

If you'd like to try out the new code, we'd be very happy to get feedback!
